### PR TITLE
add core.sys.freebsd.pthread_np

### PR DESCRIPTION
**Pull request by MartinNowak**

### CODE REVIEW

**Comment by MartinNowak** [comment]
Ready to merge?

File: src/core/sys/freebsd/pthread_np.d [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+public import core.sys.posix.time;
+
+enum __BSD_VISIBLE = true;
+
+alias pthread_switch_routine_t = void function(pthread_t, pthread_t);
+
+int pthread_attr_setcreatesuspend_np(pthread_attr_t *);
+int pthread_attr_get_np(pthread_t, pthread_attr_t *);
+int pthread_attr_getaffinity_np(const pthread_attr_t *, size_t, cpuset_t *);
+int pthread_attr_setaffinity_np(pthread_attr_t *, size_t, const cpuset_t *);
```
Comment (AndrejMitrovic) [inline src/core/sys/freebsd/pthread_np.d]: Should these const params be `const(T)*` throughout the module?

File: mak/COPY [hunk 2]
```diff
@@ -42,10 +42,13 @@ COPY=\
 	\
 	$(IMPDIR)\core\sys\freebsd\dlfcn.d \
 	$(IMPDIR)\core\sys\freebsd\execinfo.d \
+	$(IMPDIR)\core\sys\freebsd\pthread_np.d \
+	$(IMPDIR)\core\sys\freebsd\sys\_bitset.d \
```
Comment (AndrejMitrovic) [inline mak/COPY]: Why do these modules have an underscore prepended? Just to match what the C files do?

File: src/core/sys/freebsd/sys/_bitset.d [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+{
+    static if (__bitset_words!s == 1)
+        return 0;
+    else
+        return n / _BITSET_BITS;
+}
+
+struct BITSET_DEFINE(size_t s)
+{
+    c_long[__bitset_words!s] __bits;
```
Comment (AndrejMitrovic) [inline src/core/sys/freebsd/sys/_bitset.d]: Are double-underscore symbols supposed to be reserved only for the compiler or also druntime?

Comment (MartinNowak) [inline src/core/sys/freebsd/sys/_bitset.d]: It's just like in the C header.

File: mak/COPY (hunk 2 above)
Comment (MartinNowak) [inline mak/COPY]: Yep

File: src/core/sys/freebsd/pthread_np.d (hunk 1 above)
Comment (MartinNowak) [inline src/core/sys/freebsd/pthread_np.d]: Done

**Comment by DmitryOlshansky** [comment]
Rebase?

**Comment by DmitryOlshansky** [comment]
@MartinNowak ping

**Comment by DmitryOlshansky** [comment]
@MartinNowak still needs rebase.

**Comment by MartinNowak** [comment]
done @DmitryOlshansky

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**